qa/os/src/test/java/org/elasticsearch/packaging/util/Docker.java

```diff
@@ -291,6 +291,22 @@ public static void copyFromContainer(Path from, Path to) {
 
             return super.getScriptCommand("docker exec --user elasticsearch:root --tty " + containerId + " " + script);
         }
+
+        @Override
+        public Result run(String script) {
+            try {
+                return super.run(script);
+            } catch (ShellException e) {
+                try {
+                    final Shell.Result dockerLogs = getContainerLogs();
+                    throw new ShellException(
+                        e.getMessage() + "\n\nContainer stdout: " + dockerLogs.stdout + "\n\nContainer stderr: " + dockerLogs.stderr
```

Comment:
Even though we are logging the original exception message, we should still add it as the cause to this one, otherwise we lose the original call site stacktrace.